### ISSUE TITLE
Freezes and doesn't respond

### ISSUE THREAD

**Issue by prateekmedia** [jmanc3/winbar#40]
Many times when I open a start menu and try to do something then winbar freezes in the start menu open state and nothing works not even clicks or anything, It also doesn't update the app icon list section regarding what windows are open.

Then I have to kill and launch it using terminal.

I have tried to capture terminal logs this time, hope it helps:
```yaml
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not permitted (1)
[ALSOFT] (EE) Error enumerating devices: Capture Stream did not get ready (Timeout)
error: : cannot open
error: : cannot open
error: : cannot open
qt.svg: Error while inflating gzip file: SVG format check failed
qt.gui.icc: fromIccProfile: failed minimal tag size sanity
```

**Comment by jmanc3** [comment]
It's always hard to know for sure if you've fixed a bug when you can't reproduce it but I think [89ae7d2](https://github.com/jmanc3/winbar/commit/89ae7d2dedb82536e259b02501d4ac25ae659740) probably fixed it. I'll keep the issue open for a while and, if it happens again, tell me.

**Comment by prateekmedia** [comment]
@jmanc3 I found another bug that crashes this, do the following to reproduce the same:
1) Open start menu
2) Type a letter like `s`
5) Press super key

And it crashes

**Comment by jmanc3** [comment]
@prateekmedia fixed the super key crash, but I've run in to the freezing bug again. Luckily it was compiled with symbols on when it happened, so I saw what the problem was, but I have been working on icon search, so I wont get to fix it until later.

**Comment by prateekmedia** [comment]
@jmanc3 It is not launching for me with the latest commits, compiler doesn't give any error but no bar is visible.

Reverting to previous commit (89ae7d2dedb82536e259b02501d4ac25ae659740) fixes this.

**Comment by prateekmedia** [comment]
@jmanc3 It gives this error and exits
```
terminate called after throwing an instance of 'std::filesystem::__cxx11::filesystem_error'
  what():  filesystem error: cannot increment recursive directory iterator: Permission denied
```

**Comment by prateekmedia** [comment]
> @prateekmedia try the latest commit [9898a5e](https://github.com/jmanc3/winbar/commit/9898a5ef45ef23f6d9d54a4f7f41ea2c99984802)

New error, also the starting popup is annoying it would be better if its in bottom like the taskbar and with the same height
```
terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::_M_create
```

**Comment by jmanc3** [comment]
@prateekmedia the popup can be closed (it'll still cache in the background), and it only shows the first time Winbar is launched so it's fine if it's a little annoying. That string bug should be fixed in [cf17aa7](https://github.com/jmanc3/winbar/commit/cf17aa78b079e4d4767d05f834ecf3be5319a789)

**Comment by jmanc3** [comment]
Just so I don't forget to fix it, this freeze happens when we try to execute the command that gets the default interface and capture the output. For some reason, sometimes it locks up on a "wait"
